lib/Dialect/FIRRTL/Transforms/CheckCombCycles.cpp

```diff
@@ -301,7 +301,27 @@ class DummySourceNodeIterator
 public:
   explicit DummySourceNodeIterator(Node node, bool end = false)
       : node(node), connect(end ? node.context->connects.end()
-                                : node.context->connects.begin()) {}
+                                : node.context->connects.begin()) {
+    // Remove all the register self initialization connects.
+    while (connect != node.context->connects.end() &&
+           ((*connect).getSrc() == (*connect).getDest() &&
+            isFromRegOp((*connect).getDest())))
+      connect++;
+  }
+
+  bool isFromRegOp(Value v) {
+    if (v.isa<BlockArgument>())
+      return false;
+    auto defOp = v.getDefiningOp();
+    return TypeSwitch<Operation *, bool>(defOp)
+        .Case<InstanceOp>([&](InstanceOp instance) { return false; })
+        .Case<SubfieldOp>(
+            [&](SubfieldOp sub) { return isFromRegOp(sub.getInput()); })
+        .Case<SubindexOp>(
+            [&](SubindexOp sub) { return isFromRegOp(sub.getInput()); })
+        .Case<RegOp, RegResetOp>([&](auto reg) { return true; })
+        .Default([&](auto) { return false; });
+  };
```

Comment:
You can get this with `isa_and_nonnull<RegOp, RegResetOp>(getFieldRef(v).value)`.